src/hosted-native-traffic-driver/web/index.js

```diff
@@ -0,0 +1,60 @@
+import { getIframeId, getWebfonts, resizeIframeHeight, reportClicks } from '../../_shared/js/messages';
+import { enableToggles } from '../../_shared/js/ui';
+import { addTrackingPixel } from '../../_shared/js/ads';
+import { write } from '../../_shared/js/dom';
+import addColourContrastClass from '../../_shared/js/hosted-colours.js';
+import { generatePicture } from '../../_shared/js/capi-images';
+import { URLSearchParams } from '../../_shared/js/utils';
+
+const ENDPOINT = 'https://api.nextgen.guardianapps.co.uk/commercial/api/traffic-driver.json';
+
+const OVERRIDES = {
+    headline: '[%ArticleHeadline%]',
+    imageUrl: '[%ArticleImage%]',
+    imageAlt: '[%ArticleImageAlternateText%]'
+};
+
+// Loads the article data from CAPI in JSON format.
+function retrieveCapiData() {
+    if('[%ArticleShortURL%]'.length === 0) return Promise.resolve();
```

Comment:
The precompiler (`babel`) will remove that line, because it is able to infer it will always be false. I suggest you bind the DFP var to a local:

```
let articleUrl = '[%ArticleShortURL%]';
if (articleUrl.length === 0) ...
```